### PR TITLE
Implement table select user settings

### PR DESCRIPTION
**Pull request by AdrianJendo**
Sorry it's so big, I just wanted to include all the last major frontend changes.

- Implement actions for selecting users
- Other misc quality of life changes / updates

https://github.com/lynx-locks/lynx-web/assets/55325093/0b492c51-77d2-4954-a276-ef3bf1c90f48

https://github.com/lynx-locks/lynx-web/assets/55325093/0e2818f0-9773-46b1-ac9a-fee1bdc124d0

### CODE REVIEW

File: backend/controller/user.go [hunk 1]
```diff
@@ -17,20 +17,31 @@ import (
...
+	userIdsParam := r.URL.Query().Get("users")
+	if userIdsParam == "" {
+		var err error
+		err, users = dbHelpers.GetAllTable(w, []models.User{})
+		if err != nil {
+			return
+		}
+	} else {
+		// Split the comma-separated string into individual IDs
+		userIds := strings.Split(userIdsParam, ",")
```
Comment (ABarroso647) [inline backend/controller/user.go]: does this give a proper error msg on string input?

Comment (ABarroso647) [reply backend/controller/user.go]: as in like incorrect input

File: backend/controller/user.go [hunk 2]
```diff
@@ -138,14 +148,25 @@ func CreateUser(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	res := db.DB.Unscoped().Select(clause.Associations).Delete(&models.User{}, userIds.Users)
+	if res.Error != nil {
+		dbHelpers.DBErrorHandling(res.Error, w)
 		return
 	}
-	err = dbHelpers.DeleteObjandAssociationsByPk(w, models.User{Id: uId})
-	if err != nil {
+	if res.RowsAffected < 1 {
```
Comment (ABarroso647) [inline backend/controller/user.go]: should be erroring if we have less than inputted ids, no?

Comment (AdrianJendo) [reply backend/controller/user.go]: resolved

File: backend/controller/user.go [hunk 3]
```diff
@@ -167,19 +188,20 @@ func GetUserCreds(w http.ResponseWriter, r *http.Request) {
...
 	w.Header().Set("Content-Type", "application/json")
-	err, uId := helpers.ParseInt(w, r, "userId")
+	userIds := struct {
+		Users []uint `json:"users"`
+	}{}
+
+	err := json.NewDecoder(r.Body).Decode(&userIds)
 	if err != nil {
-		return
+		http.Error(w, "Malformed request", http.StatusBadRequest)
```
Comment (ABarroso647) [inline backend/controller/user.go:207]: should still return here right?

File: backend/controller/user.go (hunk 1 above)
Comment (AdrianJendo) [reply backend/controller/user.go]: If the input is incorrect it returns no users

File: backend/controller/user.go [hunk 4]
```diff
@@ -167,20 +188,21 @@ func GetUserCreds(w http.ResponseWriter, r *http.Request) {
...
 	}
-	res := db.DB.Unscoped().Select(clause.Associations).Where(models.Credential{UserId: uId}).Delete(&[]models.Credential{})
+
+	res := db.DB.Unscoped().Where("user_id IN ?", userIds.Users).Find(&models.Credential{}).Delete(&[]models.Credential{})
```
Comment (ABarroso647) [inline backend/controller/user.go]: did we stest this one multiple creds? I had an account with 3 and it only deleted one

Comment (ABarroso647) [reply backend/controller/user.go]: verified it only deleted 1/2 the second time

File: backend/controller/user.go (hunk 1 above)
Comment (ABarroso647) [reply backend/controller/user.go]: probably more ideal to actually parse it: https://stackoverflow.com/questions/24972950/go-convert-strings-in-array-to-integer 
this seems helpful

File: backend/controller/user.go [hunk 5]
```diff
@@ -17,20 +17,39 @@ import (
...
+		var err error
+		err, users = dbHelpers.GetAllTable(w, []models.User{})
+		if err != nil {
+			return
+		}
+	} else {
+		// Split the comma-separated string into individual IDs
+		userIds := make([]int, 0)
+		for _, id := range strings.Split(userIdsParam, ",") {
+			intId, err := strconv.Atoi(id)
```
Comment (ABarroso647) [inline backend/controller/user.go]: sorry I didnt realize ud paste it exactly, still need to make it uint, I think u can just wrap unit(intvalue) not 100p

Comment (AdrianJendo) [reply backend/controller/user.go]: Atoi converts it to an int, why do we need it to be a uint?

File: backend/controller/user.go [hunk 6]
```diff
@@ -167,20 +196,21 @@ func GetUserCreds(w http.ResponseWriter, r *http.Request) {
...
+	}{}
+
+	err := json.NewDecoder(r.Body).Decode(&userIds)
 	if err != nil {
-		return
+		http.Error(w, "Malformed request", http.StatusBadRequest)
 	}
-	res := db.DB.Unscoped().Select(clause.Associations).Where(models.Credential{UserId: uId}).Delete(&[]models.Credential{})
+
+	res := db.DB.Unscoped().Select(clause.Associations).Where("user_id IN ?", userIds.Users).Delete(&[]models.Credential{})
```
Comment (ABarroso647) [inline backend/controller/user.go:211]: can u paste new sql output? this looks like it'll perform better tho perhaps (well I'm assuming u tested it)

File: backend/controller/user.go (hunk 5 above)
Comment (ABarroso647) [reply backend/controller/user.go]: cuz all our ids are uints?

Comment (ABarroso647) [reply backend/controller/user.go]: if u look at the helper function that normally does this its always converting to uint

File: backend/controller/user.go (hunk 6 above)
Comment (AdrianJendo) [reply backend/controller/user.go:211]: It was DELETE * FROM credentials WHRE user_id in ()

File: backend/controller/user.go (hunk 3 above)
Comment (ABarroso647) [reply backend/controller/user.go:207]: unresolved, it does not return here

File: backend/controller/user.go (hunk 6 above)
Comment (ABarroso647) [reply backend/controller/user.go:211]: perfect, ty

File: backend/controller/user.go (hunk 3 above)
Comment (ABarroso647) [reply backend/controller/user.go:207]: can you also add it in deleteUser, that is probably my bad tho

**Review by cameronkinsella — APPROVED** [review]

**Pull request merged**